tickit/adapters/interpreters/endpoints/http_endpoint.py

```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass
+from typing import AnyStr, Callable, Generic
+
+from aiohttp.web_routedef import RouteDef
+
+
+@dataclass(frozen=True)
+class HTTPEndpoint(Generic[AnyStr]):
+    """A decorator to register a device adapter method as a HTTP Endpoint.
+
+    Args:
+        url (str): The URL that will point to a specific endpoint.
+        method (str): The method to use when using this endpoint.
+        name (str): The name of the route.
+        include_json (bool): A flag to indicate whether the route should include json.
+        interrupt (bool): A flag indicating whether calling of the method should
+            raise an adapter interrupt. Defaults to False.
+
+    Returns:
+        Callable:
+            A decorator which registers the adapter method as an endpoint.
+    """
+
+    url: str
+    method: str
+    include_json: bool = False
+    interrupt: bool = False
+
+    def __call__(self, func: Callable) -> Callable:
```

Comment:
Since we know the signature of this callable might as well restrict it:
```suggestion
    def __call__(self, func: Callable[[web.Request], web.Response]) -> Callable[[web.Request], web.Response]:
```

Ditto for `define()`